### DISCUSSION TITLE
Redis or not

### DISCUSSION THREAD

**Discussion by Jeffrey-de-Bruijn** [saleor/saleor#10278] [Deployment] [answered]
Hi, the documentation recommends Redis.

Can you please explain what difference there is in a deployment with Redis and one without?

Is Redis only used for Celery queues?

**Comment by karolkielecki** [comment, marked as answer]
Hi @Jeffrey-de-Bruijn! If you run Saleor locally, Redis is used as the broker for Celery and as Cache. Without Redis, you have only an in-memory cache. Tasks are executed immediately while the request is being processed instead of async. For running Saleor locally and if you want to test something, that's fine to go without Redis.

However, it shouldn't work like that for production purposes. A celery broker is required, but it doesn't have to be Redis. It can be Rabbit or AWS SQA.

On the other hand, running Saleor without Cache is possible, but it will be less efficient.

**Reply by Jeffrey-de-Bruijn** [reply]
If I wanted to scale horizontally and run 2 or more containers of Saleor's core, how do these containers behave with Redis? Is this a possibility that is accounted for?

**Reply by karolkielecki** [reply]
Yes, that's what Redis does. Redis can work with many workers at the same time.
